janus.c

```diff
@@ -2668,12 +2679,45 @@ int janus_process_incoming_admin_request(janus_request *request) {
 				janus_events_notify_handlers(JANUS_EVENT_TYPE_SESSION, JANUS_EVENT_SUBTYPE_NONE,
 					session_id, "destroyed", NULL);
 			goto jsondone;
-		}
-		/* If this is not a request to destroy a session, it must be a request to list the handles */
-		if(strcasecmp(message_text, "list_handles")) {
+		} else if (!strcasecmp(message_text, "set_session_timeout")) {
+			/* Specific session timeout setting. */
+			JANUS_VALIDATE_JSON_OBJECT(root, session_timeout_parameters,
+				error_code, error_cause, FALSE,
+				JANUS_ERROR_MISSING_MANDATORY_ELEMENT, JANUS_ERROR_INVALID_ELEMENT_TYPE);
+			if(error_code != 0) {
+				ret = janus_process_error_string(request, session_id, transaction, error_code, error_cause);
```

Comment:
I got a warning here when compiling:

```
janus.c:2688:59: warning: passing argument 3 of ‘janus_process_error_string’ from incompatible pointer type [-Wincompatible-pointer-types]
 2688 |     ret = janus_process_error_string(request, session_id, transaction, error_code, error_cause);
      |                                                           ^~~~~~~~~~~
      |                                                           |
      |                                                           json_t * {aka struct json_t *}
janus.c:89:96: note: expected ‘const char *’ but argument is of type ‘json_t *’ {aka ‘struct json_t *’}
```

This should indeed be `transaction_text` and not `transaction` (which is a jansson object).